rust/compiler/src/parser/mod.rs

```diff
@@ -259,25 +266,26 @@ pub fn prefix_annotation_(i: Input) -> Res<Input, (adlast::ScopedName, serde_jso
 
 pub fn merge_annotations(
     anns: Vec<(adlast::ScopedName, serde_json::Value)>,
-) -> adlast::Annotations {
-    // Create a map out of the annotations, but join any doc strings as separate lines
+) -> Result<adlast::Annotations,AnnotationError> {
```

Comment:
Returning a custom error here requires more boilerplate elsewhere, and it's not really necessary. It can only fail one way, no it might as well just have the error type be a String, ie:

```
pub fn merge_annotations(
    anns: Vec<(adlast::ScopedName, serde_json::Value)>,
) -> Result<adlast::Annotations, String> {
```